core/sdk-core/src/main/java/software/amazon/awssdk/core/retry/RetryPolicy.java

```diff
@@ -62,8 +62,8 @@ private RetryPolicy(BuilderImpl builder) {
         this.numRetries = builder.numRetries;
         this.retryCondition = builder.retryCondition;
         this.retryCapacityCondition = builder.retryCapacityCondition;
-
         this.aggregateRetryCondition = generateAggregateRetryCondition();
+        this.fastFailRateLimiting = builder.isFastFailRateLimiting();
```

Comment:
Should we throw validation exception if this is provided when retry mode is not adaptive?

Reply:
Done. Switched to the nullable `Boolean` as well